daemon/cmd/daemon_main.go

```diff
@@ -1704,7 +1704,15 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], promise.Pr
 	var wg sync.WaitGroup
 
 	params.Lifecycle.Append(cell.Hook{
-		OnStart: func(cell.HookContext) error {
+		OnStart: func(cell.HookContext) (err error) {
+			defer func() {
+				// Reject promises on error
+				if err != nil {
```

Comment:
Does this still work if you shadow `err`? like in `d, restoredEndpoints, err := newDaemon(daemonCtx, cleaner, &params)`. 

As far as I know, the moment you shadow declare with `:=` the `err` in the defer and that below the new declaration are not the same anymore.

Reply:
They'll be the same if returned, I think, i.e., return assigns to the return value `err`, and that is what we are inspecting here. Works for cases like `return nil`, etc.